International Classes Support

dotpy3 opened the pull request:
See [issue 12](https://github.com/uvweb/UVweb/issues/12).

MiLk commented on src/Uvweb/UvBundle/Controller/DetailController.php:
```diff
@@ -18,7 +18,7 @@ public function __construct()
     }
 
-    public function detailAction($uvname = '')
+    public function detailAction($uvname = '', $uniId = '')
```
Can't we have `null` as a default value for `$uniId`?

MiLk commented on src/Uvweb/UvBundle/Controller/DetailController.php:
```diff
@@ -87,7 +96,7 @@ public function detailAction($uvname = '')
     }
 
-    public function postAction($uvname) 
+    public function postAction($uvname, $uniId = '') 
```
Can't we have null as a default value for `$uniId`?

MiLk commented on src/Uvweb/UvBundle/Controller/HomeController.php:
```diff
@@ -90,10 +90,12 @@ public function getUvNamesLikeAction()
 
-        return new Response(json_encode($uvNames));
+        $resp = new Response(json_encode($uvNames));
+        $resp->headers->set('Content-Type', 'application/json; charset=utf-8');
```
Couldn't it be the third argument of the Response contructor?
It would be easier to read/maintain.

MiLk commented on src/Uvweb/UvBundle/Controller/UnisController.php:
```diff
@@ -0,0 +1,257 @@
+          $unis = $uniRepository->getWithMinimalInfo();
+
+          return $this->render('UvwebUvBundle:Unis:all_ordered_by_country.html.twig', array('countries' => $unis, 'order' => $order));
+        }
```
If `$order` is not `country` what happens?

MiLk commented on src/Uvweb/UvBundle/Controller/UnisController.php:
```diff
@@ -0,0 +1,257 @@
+
+            //return new Response('<html><body>ok</body></html>');
+
+            if($order === 'dynamic') //Dynamic: pass the UV names to the appropriate view
```
Please add curly braces around the next line.

MiLk commented on src/Uvweb/UvBundle/Entity/UniversityRepository.php:
```diff
@@ -0,0 +1,62 @@
+
+    $sortedUniversities = array();
+    foreach($universities as $university) {
+      if (is_null($university["num_comments"])) $university["num_comments"] = 0;
```
AFAIK all of the conditions here could be replaced by SQL functions.
See https://github.com/uvweb/UVweb/issues/12#issuecomment-266763416

dotpy3 replied:
Thanks, that's been taking care of.

dotpy3 commented:
Thanks, I've pushed commit `90ee5b7` that fixes:

+ **Profile**: comments on a profile are now correctly displayed and provide link to the correct UV.

+ **Ordering and sorting**: now removed from the router for universities, waiting for someone to implement it (in the same way as UVs can be sorted).

+ **SQL requests**: fixed accordingly - moved part of the logic from the PHP code to the SQL requests as requested by @MiLk.

Last commit `4cb015d` also fixes the web service requests - international classes now cannot be accessed via the web services, given that the apps don't implement them.

MiLk commented on src/Uvweb/UvBundle/Controller/DetailController.php:
```diff
@@ -43,8 +43,16 @@ public function detailAction($uvname = '')
+            $uni = null;
+        } else
+        {
+            $uni = $this->getDoctrine()->getManager()->getRepository('UvwebUvBundle:University')->find($uniId);
```
What about something more compact like:
```php
$uni = is_null($unidId) ? null : $this->getDoctrine()->getManager()->getRepository('UvwebUvBundle:University')->find($uniId);
```

MiLk commented:
Codewise, LGTM

dotpy3 commented:
Thanks. I've also fixed a last bug that I didn't see, and rebased the PR in a single commit. If that's good for you, I think we can merge.

MiLk commented:
Ask them to backup the database before doing the migration.
I don't remember if there is automated backups or not.
